pemFioi/quickAlgo/interface-mobileFirst.js

```diff
@@ -249,29 +271,122 @@ var quickAlgoInterface = {
+            "    <div class=\"panel-heading\">" +
+            "        <h2 class=\"sectionTitle\">" +
+            "            <span class=\"iconTag\"><i class=\"icon fas fa-pencil-alt\"></i></span>" +
+                         this.strings.editWindowTitle +
+            "        </h2>" +
+            "    <div class=\"exit\" id=\"editclose\"><i class=\"icon fas fa-times\"></i></div>" +
+            "    </div>" +
+            "    <div class=\"panel-body\">" +
+            "        <div id=\"editExerciseTitle\">" +
+            "            <label>" + this.strings.titleEdition + "</label><input id=\"editExerciseTitleInput\" type=\"text\" value=\"" + title + "\"/>" +
+            "        </div>" +
+            "        <div id=\"editExerciseDescription\">" +
+            "            <label>" + this.strings.descriptionEdition + "</label>" +
+            "            <textarea rows=\"10\" id=\"editExerciseDescriptionTextarea\">" + subject + "</textarea>" +
+            "        </div>" +
+            "        <div id='panel-body-bottom'>" +
+            "            <button id='saveExerciseChanges'>" + this.strings.saveAndQuit + "</button>" +
+            "        </div>" +
+            "    </div>" +
+            "</div>";
+
+        window.displayHelper.showPopupDialog(editExerciseHtml);
+
+        var that = this;
+
+        $("#editclose").click(function() {
+            var newTitle = $("#editExerciseTitleInput").val();
+            var newDesc = $("#editExerciseDescriptionTextarea").val();
+            var oldTitle = document.title;
+            var oldDescription = $(".exerciseText").first().text();
+            if (newTitle !== oldTitle || newDesc !== oldDescription) {
+                if (!window.confirm(that.strings.quitWithoutSavingConfirmation)) {
+                    return;
+                }
+            }
+
+            $('#popupMessage').hide();
+            window.displayHelper.popupMessageShown = false;
+        });
+
+        $("#saveExerciseChanges").click(function() {
+            $('#popupMessage').hide();
+            window.displayHelper.popupMessageShown = false;
+
+            var newTitle = $("#editExerciseTitleInput").val();
+            var newSubject = $("#editExerciseDescriptionTextarea").val();
+            that.userTaskData.title = newTitle;
+            that.userTaskData.subject = newSubject;
+            that.loadSubjectFromUserTaskData();
+        });
+    },
+
     loadPrograms: function(formElement) {
         this.blocklyHelper.handleFiles(formElement.files);
         resetFormElement($(formElement));
         this.closeEditorMenu();
     },
 
+    /**
+     * This function allow us to save the subject into the additional data saved inside of the interface
+     * This also save the element from the context
+     * @param additional The additional data where we should save subject
+     */
+    saveAdditional: function(additional) {
```

Comment:
I have renamed this function which was named (save subject) in precedent pull request, but the function below was "loadAdditional", so it has more sense to rename this function "saveAdditional" in order to keep the same name as the one below